cmd/cluster/support/post_test.go

```diff
@@ -0,0 +1,115 @@
+package support
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osdctl/internal/support"
+)
+
+func TestAccessFile(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	testCases := []struct {
+		title       string
+		path        string
+		errExpected bool
+	}{
+		{
+			title:       "File not existant",
+			path:        "/tmp/fakeFile.json",
+			errExpected: true,
+		},
+		{
+			title:       "File existant",
+			path:        "./post.go",
+			errExpected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.title, func(t *testing.T) {
+			_, err := accessFile(tc.path)
+			if tc.errExpected {
+				g.Expect(err).Should(HaveOccurred())
+			} else {
+				g.Expect(err).ShouldNot(HaveOccurred())
+			}
+		})
+	}
+}
+
+func TestValidateBadResponse(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	//good json format
+	gMsg := "{\"field1\":\"data1\",\"field2\":\"data2\"}"
+
+	//bad json format
+	bMsg := "{\"field1\":\"data1\",\"field2\"\"data2\"}"
+
+	testCases := []struct {
+		title       string
+		body        []byte
+		errExpected bool
+	}{
+		{
+			title:       "Validate good Json Response",
+			body:        []byte(gMsg),
+			errExpected: false,
+		},
+		{
+			title:       "Validate bad Json Response",
+			body:        []byte(bMsg),
+			errExpected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.title, func(t *testing.T) {
+			_, err := validateBadResponse(tc.body)
+			if tc.errExpected {
+				g.Expect(err).Should(HaveOccurred())
+			} else {
+				g.Expect(err).ShouldNot(HaveOccurred())
+			}
+		})
+	}
+}
+
+func TestValidateGoodResponse(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	//good json format
+	gMsg := "{\"field1\":\"data1\",\"field2\":\"data2\"}"
+	//bad json format
+	bMsg := "{\"field1\":\"data1\",\"field2\"\"data2\"}"
+
+	testCases := []struct {
+		title        string
+		body         []byte
+		lmtSprReason support.LimitedSupport
+		errExpected  bool
```

Comment:
Same as above, since there are two different errors returned from the function, making sure that the error returned is the one expected may be good to add here.

Reply:
Same as above, this function returns two different errors